apisix/init.lua

```diff
@@ -267,6 +267,11 @@ local function set_upstream_headers(api_ctx, picked_server)
         api_ctx.var.var_x_forwarded_host = x_forwarded_host
     end
 
+    local x_forwarded_for = api_ctx.var.proxy_add_x_forwarded_for
+    if x_forwarded_for then
+        api_ctx.var.var_x_forwarded_for = x_forwarded_for
```

Comment:
Maybe we can set `proxy_set_header   X-Forwarded-For      $proxy_add_x_forwarded_for` directly?

Reply:
so we don't need `var_x_forwarded_for`?

Reply:
Yes. It seems there are no builtin replacements for the others.

Reply:
Done